File: 01_Leetcode_FAANG/Swagat_Shekhar/Number_of_Good_Pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int n = nums.length;
        int count=0;
        for(int i=0;i<n;i++){
            for(int j=i;j<n;j++){
                if(nums[i]==nums[j] && i<j){
                    count++;
                }
            }
        }
        return count;
    }
}

File: 01_Leetcode_FAANG/S_S_Nanda/Number_of_good_pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        unordered_map<int, int> count;
        int ans=0;
        
        for(auto it: nums)
        {
            ans += count[it];
            count[it]++;
        }
        return ans;
    }
};
